scripts/check-go-version.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/bash
+
+version_regex="[0-9]\.[0-9]"
+update=false
+
+while getopts "u" opt; do
+    case $opt in
+        u) update=true ;;
+        *) echo "Usage: $0 [-u]" >&2
+           exit 1
+           ;;
+    esac
+done
+
+# Fetch latest go release version
+go_latest_version=$(curl -s https://go.dev/dl/?mode=json | jq -r '.[0].version' | awk -F'.' '{gsub("go", ""); print $1"."$2}')
+
+# Extract go version from go.mod
+go_mod_path="./go.mod"
+go_mod_version=$(grep -oP "go\s+\K$version_regex+" $go_mod_path)
+
+# If update, set go.mod version as latest version - 1 (N-1)
+if [ "$update" = true ]; then
+    go_mod_updated="${go_latest_version%.*}.$((10#${go_latest_version#*.} - 1))"
+    if [ "$go_mod_version" != "$go_mod_updated" ]; then
+        sed -i "s/$go_mod_version/$go_mod_updated/g" "$go_mod_path"
+        {
+            printf "Go.mod file: %-60s | Previous Go Version: %s | Updated Go Version: %s\n" "$go_mod_path" "${go_mod_version}" "${go_mod_updated}"
+        } | column -t -s '|'
+    fi
+fi
+
+# Extract go version from GitHub actions scripts
+while IFS=: read -r build_script version_string; do
+    
+    # If update, set latest go version for all GitHub actions scripts
+    if [ "$update" = true ]; then
+        if [ "${version_string##*: }" != "$go_latest_version" ]; then
+            updated_version="${version_string%%:*}: ${go_latest_version}"
+            sed -i "s/$version_string/$updated_version/g" "$build_script"
```

Comment:
how do you know this will not update some other part of the file? E.g. some GH actions are included via hash, which could in theory match the string you're substituting